src/lib/shopify/storefront/provider.tsx

```diff
@@ -47,6 +49,7 @@ const InternalContextProvider: React.FC<InternalContextProviderProps> = ({
   client,
   appCartId
 }) => {
+  const events = new EventEmitter()
```

Comment:
i think this should be outside the component to prevent new instances on each render.